### PR TITLE
Add generated code check

### PR DESCRIPTION
**Pull request by pakrym**
Adds 3 checks:

1. All sln files reference existing projects (for track2 solutions)
2. File that include code snippets is up to date
3. Storage code is up to date.

### CODE REVIEW

**Comment by JoshLove-msft** [comment]
For Storage generated code being up to date, can we also check that generate.ps1 doesn't error out? If it errors out, the generated code will not be updated the diff check would pass (this happened recently).

**Comment by pakrym** [comment]
> For Storage generated code being up to date, can we also check that generate.ps1 doesn't error out? If it errors out, the generated code will not be updated the diff check would pass (this happened recently).

Can you update the generate.ps1 so it returns non-0 exit code when generation fails? 

CodeCheck already checks for that.

**Comment by JoshLove-msft** [comment]
> > For Storage generated code being up to date, can we also check that generate.ps1 doesn't error out? If it errors out, the generated code will not be updated the diff check would pass (this happened recently).
> 
> Can you update the generate.ps1 so it returns non-0 exit code when generation fails?
> 
> CodeCheck already checks for that.

Ah ok, I missed that.

File: eng/CodeChecks.ps1 [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    Invoke-Block {
+        & $repoRoot\storage\generate.ps1
+    }
+
+    Write-Host "git diff"
+    & git diff --ignore-space-at-eol --exit-code
+    if ($LastExitCode -ne 0) {
+        $status = git status -s | Out-String
+        $status = $status -replace "`n","`n    "
+        LogError "Generated code is not up to date."
```
Comment (tg-msft) [inline eng/CodeChecks.ps1]: Nit - can you do this check after each section and then add more detailed errors like can you make this `"Generated Storage code is not up to date (please re-run sdk\storage\generate.ps1)."` to make it easy for folks to solve any issues themselves?

**Review by tg-msft — APPROVED** [review]
Thanks for fixing this!

File: eng/CodeChecks.ps1 (hunk 1 above)
Comment (pakrym) [reply eng/CodeChecks.ps1]: I've added more info to the message itself, let's see if people would get confused by it and add more granular failures if needed.

**Comment by tg-msft** [comment]
Btw, `generate.ps1` assumes `npm` is installed on the machine.  Do we need to set anything up for that?

**Comment by pakrym** [comment]
@tg-msft does the current failure in the build log looks like anything to you?

**Comment by tg-msft** [comment]
Might be missing autorest or ts?  We might need to update package.json to include them as dev dependencies and then have generate.ps1 install them.

**Comment by pakrym** [comment]
Failure https://github.com/Azure/azure-sdk-for-net/issues/8590

**Pull request merged**